src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheck.java

```diff
@@ -255,11 +267,14 @@ private static boolean isInNoBraceControlStatement(DetailAST ast) {
     }
 
     /**
-     * Tests whether the given AST is used in the "assignment in while" idiom.
+     * Tests whether the given AST is used in the "assignment in loop" idiom.
      * <pre>
      * String line;
      * while ((line = bufferedReader.readLine()) != null) {
-     *    // process the line
+     *   // process the line
+     * }
+     * for (;(line = bufferedReader.readLine()) != null;) {
```

Comment:
The opposite of above. Why not make both examples the same?

Also why did we add the for loop example but not the do-while?

Reply:
> Also why did we add the for loop example but not the do-while?

There isn't a specific reason, I thought this would be enough for explanation purposes, I'll add do-while example too.




> The opposite of above. Why not make both examples the same?

My apologies, I couldn't understand this remark. Should I add a different kind of example for each type of `loop` or I just use any one loop as an example?

Reply:
the same code is used